### PR TITLE
fix(report-add): make to add error toast message when already attached report is added again into dashboard or chart

### PR DESCRIPTION
**Pull request by prosdev0107**
### SUMMARY
Attach new report to the dashboard again, but there are no error messages on the page #18225

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
BEFORE:
https://user-images.githubusercontent.com/47900232/158206908-7a8fe15e-31cf-4049-b7db-395d0755cecf.mov
Screen Shot 2022-03-14 at 11 37 26 AM

AFTER:
image

### TESTING INSTRUCTIONS
**How to reproduce the bug**

1. Create a report for a chart or dashboard from the top right settings menu "Alerts & Reports" section - make sure you create a Report and not an alert
2. Go to the chart or dashboard that you created the report for
3. Click the calendar icon to schedule a report

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/components/ReportModal/index.tsx [hunk 1]
```diff
@@ -250,9 +251,16 @@ const ReportModal: FunctionComponent<ReportProps> = ({
         await dispatch(addReport(newReportValues as ReportObject));
         onHide();
       } catch (e) {
-        const parsedError = await getClientErrorObject(e);
-        const errorMessage = parsedError.message;
-        onReducerChange(ActionType.error, errorMessage);
+        const { error } = await getClientErrorObject(e);
+        if (error) {
```
Comment (diegomedina248) [inline superset-frontend/src/components/ReportModal/index.tsx:255]: It seems the code changed from the time the issue was created.
The PR that does so is here: https://github.com/apache/superset/pull/17988

Looking at the PR, it seems the idea is to show the message in place, as opposed to use the toast.
Plus, looking at the change here, we're now assuming the error will always be a "report already exists" one, which is not correct.

I would suggest trying the following:
```js
const errorMessage = parsedError?.message || parsedError?.error;
```

If the error is not there, or if it's empty, we should use a default message I believe.
@rusackas thoughts here?

**Review by diegomedina248 — CHANGES_REQUESTED** [review]

File: superset-frontend/src/components/ReportModal/index.tsx (hunk 1 above)
Comment (prosdev0107) [reply superset-frontend/src/components/ReportModal/index.tsx:255]: @diegomedina248 @rusackas 

what is your correct expectation?

Comment (rusackas) [reply superset-frontend/src/components/ReportModal/index.tsx:255]: For now, let's put the error at the bottom, like we do on the database connection modal. Consistency above all else!

image

File: superset-frontend/src/components/ReportModal/index.tsx [hunk 2]
```diff
@@ -338,7 +351,7 @@ const ReportModal: FunctionComponent<ReportProps> = ({
                 value: target.value,
               }),
           }}
-          errorMessage={currentReport?.error || ''}
+          errorMessage=""
```
Comment (diegomedina248) [inline superset-frontend/src/components/ReportModal/index.tsx:354]: maybe this can be removed?

**Review by diegomedina248 — APPROVED** [review]

**Review by diegomedina248 — APPROVED** [review]

**Review by rusackas — APPROVED** [review]

**Pull request merged**